src/main/java/net/minecraftforge/event/TickEvent.java

```diff
@@ -66,6 +67,25 @@ public WorldTickEvent(LogicalSide side, Phase phase, Level world)
             this.world = world;
         }
     }
+
+    /**
+     * Called before and after a chunk is ticked for things like random ticks and weather. Useful for random-tick and weather based actions.
+     */
+    public static class EnvironmentTickEvent extends TickEvent
+    {
+        public final Level level;
+        public final LevelChunk chunk;
+        public final int randomTickSpeed;
+
+        public EnvironmentTickEvent(LogicalSide side, Phase phase, Level level, LevelChunk chunk, int randomTickSpeed)
+        {
+            super(Type.ENVIRONMENT, side, phase);
```

Comment:
Will it ever be possible for this event to be fired on the logical client? If not, it may be good to signal this by hardcoding `LogicalSide.SERVER` (possibly alongside the change to `ServerLevel` as suggested above).